tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -721,16 +721,23 @@ static int handleTestInvocation(TestOptions Opts, TestOptions &InitOpts) {
     break;
   }
 
-#define SEMANTIC_REFACTORING(KIND, NAME, ID) case SourceKitRequest::KIND:                 \
-    {                                                                                     \
-      sourcekitd_request_dictionary_set_uid(Req, KeyRequest, RequestSemanticRefactoring); \
-      sourcekitd_request_dictionary_set_uid(Req, KeyActionUID, KindRefactoring##KIND);    \
-      sourcekitd_request_dictionary_set_string(Req, KeyName, Opts.Name.c_str());          \
-      sourcekitd_request_dictionary_set_int64(Req, KeyLine, Opts.Line);                   \
-      sourcekitd_request_dictionary_set_int64(Req, KeyColumn, Opts.Col);                  \
-      sourcekitd_request_dictionary_set_int64(Req, KeyLength, Opts.Length);               \
-      break;                                                                              \
-    }
+#define SEMANTIC_REFACTORING(KIND, NAME, ID)                                   \
```

Comment:
Could move most of this out into a function and then just have the macro be:
```
case SourceKitRequest::KIND:
  setRefactoringFields(Req, Opts, KindRefactoring##KIND);
  break;
```

Reply:
Good idea. I also added a test case for the offset-based refactoring.

Reply:
Thanks, LGTM!